pdctl/command/global.go

```diff
@@ -111,6 +118,27 @@ func printResponseError(r *http.Response) {
 	io.Copy(os.Stdout, r.Body)
 }
 
+func validPDAddr(pd string) error {
+	u, err := url.Parse(pd)
+	if err != nil {
+		return err
+	}
+	if u.Scheme == "" {
+		u.Scheme = "http"
+	}
+	addr := u.String()
+	reps, err := http.Get(fmt.Sprintf("%s/%s", addr, pingPrifix))
+	if err != nil {
+		return err
+	}
+	defer reps.Body.Close()
+	if reps.StatusCode != http.StatusOK {
+		ioutil.ReadAll(reps.Body)
```

Comment:
If not read, the connection FD will not close even you call body close later. I remember this is a pitfall before, but don't know go improves it or not.   

Btw, We should call `ioutil.ReadAll(reps.Body)`  before the status check.